sw/device/lib/testing/mock_mmio.h

```diff
@@ -302,6 +318,37 @@ class MmioTest {
 #define EXPECT_READ32(offset, ...) \
   EXPECT_READ32_AT(this->dev(), offset, __VA_ARGS__)
 
+/**
+ * Expect a read at the given offset, returning a 32-bit value that has
+ * one bit set to 1 and all other bits set to 0.
+ *
+ * This macro receives the index of the bit which is expected to be set to 1.
+ *
+ * This function is only available in tests using a fixture that derives
+ * `MmioTest`.
+ *
+ * This expectation is sequenced with all other `EXPECT_READ` and `EXPECT_WRITE`
+ * calls.
+ */
+#define EXPECT_READ32_BIT(offset, ...) \
+  EXPECT_READ32_AT(this->dev(), offset, mock_mmio::ToIntBit(__VA_ARGS__))
+
+/**
+ * Expect a read at the given offset, returning a 32-bit value that has
+ * one bit set to 0 and all other bits set to 1.
+ *
+ * This macro receives the index of the bit which is expected to be set to 0.
+ *
+ * This function is only available in tests using a fixture that derives
+ * `MmioTest`.
+ *
+ * This expectation is sequenced with all other `EXPECT_READ` and `EXPECT_WRITE`
+ * calls.
+ */
+#define EXPECT_READ32_BIT_INVERTED(offset, ...) \
+  EXPECT_READ32_AT(this->dev(), offset,         \
+                   mock_mmio::ToIntBitInverted(__VA_ARGS__))
+
```

Comment:
These macros feel a bit overkill? It would be much more natural to write
```cc
EXPECT_READ32(..., Bit(index))
```
or similar. Compare to how we use `EXPECT_THAT`:
```
EXPECT_THAT(lhs, MyMatcher(param))
```
Given that we're imitating googletest here, I think it's prudent to not add more macros than are strictly necessary.

Reply:
I think the macros here should be dropped, they're just hiding something that has a fairly straightforward use.